src/components/instructor/AssignmentsView.js

```diff
@@ -10,15 +17,137 @@ import {useLocation} from 'react-router-dom'
 // assignment id, title, dueDate and buttons to grade, edit, delete each assignment
 
 const AssignmentsView = (props) => {
+    const headers = ['Id', 'Title', 'Due Date', 'Course Id', 'Sec Id', 'Sec No', '', ''];
+    const [asgnmts, setAsgnmts] = useState([
+
+    ]);
+    const [message, setMessage] = useState('');
+    const [search, setSearch] = useState({id: '', title: '', dueDate: '', secId: '', secNo: ''});
 
     const location = useLocation();
-    const {secNo, courseId, secId} = location.state;
-     
-    return(
-        <> 
-           <h3>Not implemented</h3>
+    const { secNo, courseId, secId } = location.state;
+
+    const fetchAsgnmts = async () => {
+        // TODO: get the assignments
+        try{
+            const response = await fetch(`${SERVER_URL}/sections/${secNo}/assignments`);
+            if (response.ok){
+                const asgnmts = await response.json();
+                setAsgnmts(asgnmts);
+            } else {
+                const json = await response.json();
+                setMessage("response error: "+ json.message);
+            }
+        } catch (err) {
+            setMessage("network error: "+ err);
+        }
+    }
+
+    useEffect( () => {
+        fetchAsgnmts();
+    }, []);
+
+    const gradeAsgnmt = async (asgnmt) => {
+        try {
+            const response = await fetch(`${SERVER_URL}/grades`,
+                {
+                    method: 'PUT',
+                    headers: {
+                        'Content-Type': 'application/json',
+                    },
+                    body: JSON.stringify(asgnmt),
+                });
+            if (response.ok) {
+                setMessage("Grade Saved")
+                fetchAsgnmts();
+            } else {
+                const rc = await response.json();
+                setMessage(rc.message);
+            }
+        } catch (err) {
+            setMessage("network error: "+err);
+        }
+    }
+
+    const editChange = (event) => {
+        setSearch({...search,  [event.target.name]:event.target.value});
+    }
```

Comment:
I don't believe these are ever used, it seems all of their functionality is handled in AssignmentGrade.js. Although this is a visual observation, I haven't tested it.

Reply:
Good catch. Removed the unused code.